src/cattr/converters.py

```diff
@@ -318,6 +318,10 @@ def structure_attrs_fromdict(self, obj, cl):
             if name[0] == "_":
                 name = name[1:]
 
+            if val is None and type_ is not None:
+                raise ValueError('property "{:s}" of type "{:s}" must not be None'.format(
+                    src_key, str(type_)))
+
```

Comment:
Would this fix #53? And would it handle `Optional[str]`? I think it should be a separate PR though, since there's a bit of uncertainty as to what the best approach is, see #26